python/paddle/base/dygraph/tensor_patch_methods.py

```diff
@@ -383,6 +383,62 @@ def gradient(self):
             return (np.array(self.grad), np.array(self.grad.rows()))
         return np.array(self.grad)
 
+    @framework.dygraph_only
+    def apply_(self, func):
+        """
+        Inplace apply the python function to the tensor.
+
+        Returns:
+            None
+
+        Examples:
+            .. code-block:: python
+
+                >>> import paddle
+
+                >>> x = paddle.to_tensor(5.)
+                >>> f = lambda x: 3*x+2
+                >>> x.apply_(f)
+                >>> print(x)
+                Tensor(shape=[], dtype=float32, place=Place(cpu), stop_gradient=True,
+                 17.)
```

Comment:
can the shape of the input data in example code be larger, such as [3, 3]

Reply:
and use tensor in GPU

Reply:
are we support other data types? eg `float16`, `bfloat16` and `float64`, should add more case in example code

Reply:
same issue in example code in `def apply` below

Reply:
done add the related example code